test/services/credential-service/credentials.test.js

```diff
@@ -231,40 +227,36 @@ describe('Credential service tests', function () {
     });
 
     it('should delete all credentials associated with a user when user is deleted', function (done) {
-      Promise.all([ credentialService.getCredential(user.username, 'oauth2'),
-        credentialService.getCredential(user.username, 'basic-auth') ])
-      .then(([oauthRes, basicAuthRes]) => {
-        should.exist(oauthRes); // Check to confirm the credentials exist
-        should.exist(basicAuthRes);
-        return userService.remove(user.id)
-        .then(res => {
-          should.exist(res);
-          res.should.eql(true);
-          return Promise.all([ credentialService.getCredential(user.username, 'oauth2'),
-            credentialService.getCredential(user.username, 'basic-auth') ])
-          .then(([oauthResAfterDelete, basicAuthResAfterDelete]) => {
-            should.not.exist(oauthResAfterDelete);
-            should.not.exist(basicAuthResAfterDelete);
-            done();
-          });
+      Promise.all([credentialService.getCredential(user.username, 'oauth2'),
+        credentialService.getCredential(user.username, 'basic-auth')])
+        .then(([oauthRes, basicAuthRes]) => {
+          should.exist(oauthRes); // Check to confirm the credentials exist
+          should.exist(basicAuthRes);
+          return userService.remove(user.id)
+            .then(res => {
+              should.exist(res);
+              return Promise.all([credentialService.getCredential(user.username, 'oauth2'),
+                credentialService.getCredential(user.username, 'basic-auth')])
+                .then(([oauthResAfterDelete, basicAuthResAfterDelete]) => {
+                  should.not.exist(oauthResAfterDelete);
+                  should.not.exist(basicAuthResAfterDelete);
+                  done();
+                });
+            });
         });
-      });
     });
 
     it('should delete a credential', function (done) {
       credentialService.insertCredential(user.username, 'oauth2')
-      .then(res => {
-        should.exist(res);
-        credentialService.removeCredential(user.username, 'oauth2')
-        .then(deleted => {
-          deleted.should.eql(true);
```

Comment:
you removed this check and check for  `resAfterDelete`, why ? 
what if some code relies on it returning true\false\undefined ?

Reply:
Good point, better to restore those.